WordPress/gutenberg
# [RNMobile] Rename example iOS project to GutenbergDemo

After merging https://github.com/WordPress/gutenberg/pull/25532, I've noticed an issue where Xcode would sometimes get confused between `gutenberg` demo target and `Gutenberg` bridge Framework.

It would build successfully from scratch and fail on subsequent builds after modifications in the project.

On this PR we try to change the whole project name to `GutenbergDemo`. It's more descriptive and won't have problems with `Gutenberg` framework imported.

*To test:*
- Checkout
- `rm -rf packages/react-native-editor/ios/Pods   packages/react-native-editor/ios/build`.
- Open `packages/react-native-editor/ios/GutenbergDemo.xcworkspace`.
- `Clean build folder` from Xcode.
- `npm install`
- `npm run native ios`.
  - Check that the project build and run on the simulator as expected.
- Build and Run from Xcode.
  - Check that it works as expected.
- Modify a .swift file on GutenbergDemo project.
- Build and Run again.
  - Check that everything works as expected.
